src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -8337,6 +8337,10 @@ namespace VariableSpeedCoils {
             state.dataLoopNodes->Node(AirOutletNode).CO2 = state.dataLoopNodes->Node(AirInletNode).CO2;
         }
 
+        if (state.dataContaminantBalance->Contaminant.GenericContamSimulation) {
+            state.dataLoopNodes->Node(AirOutletNode).GenContam = state.dataLoopNodes->Node(AirInletNode).GenContam;
+        }
```

Comment:
@Myoldmopar There are ifs like this scattered throughout the code. For simple assignments like this are these ifs saving any time or are they just clutter?

Reply:
I agree these checks aren't needed (just pass the 0). Also thinking of a safeAirNodeCopy function passing MassFlowRate, Quality, Press, MassFlowRateMin, MassFlowRateMax, MassFlowRateMinAvail and MassFlowRateMaxAvail, and CO2 and GenContam. A lot of these child update functions would reduce to:

        safeAirNodeCopy(state, outletnode, inletnode);
        state.dataLoopNodes->Node(AirOutletNode).Temp = varSpeedCoil.OutletAirDBTemp;
        state.dataLoopNodes->Node(AirOutletNode).HumRat = varSpeedCoil.OutletAirHumRat;
        state.dataLoopNodes->Node(AirOutletNode).Enthalpy = varSpeedCoil.OutletAirEnthalpy;